src/package/mod.rs

```diff
@@ -81,19 +163,92 @@ pub struct App {
     pub output_root: PathBuf,
     /// Absolute path to the main reactor file.
     pub main_reactor: PathBuf,
+    /// main reactor name
+    pub main_reactor_name: String,
     /// target language of this lf program
     pub target: TargetLanguage,
     /// platform for which this program should be compiled
     pub platform: Platform,
+    /// target properties of that lingua-franca app
+    pub properties: AppTargetProperties,
+}
 
-    pub dependencies: HashMap<String, DetailedDependency>,
-    pub properties: HashMap<String, serde_json::Value>,
+impl AppFile {
+    const DEFAULT_MAIN_REACTOR_RELPATH: &'static str = "src/Main.lf";
```

Comment:
I'm assuming we're not trying to be compatible with Windows at any level. Is that correct?

Reply:
I dont see it as a big priority.

Reply:
I'm not saying it should be, but it's something to communicate to users.

Reply:
Yes, good idea.

Reply:
done